cmd/manager/main.go

```diff
@@ -88,6 +90,26 @@ func main() {
 
 	printVersion()
 
+	// Take exclusive instance lock if required, we need that to make sure that
+	// we don't have more than one working instance of k8s-nmstate
+	if lockFilePath, ok := os.LookupEnv("NMSTATE_INSTANCE_NODE_LOCK_FILE"); ok {
+		log.Info("Try to take exclusive lock on file: %s", lockFilePath)
+		lockFileMutex, err := lockfile.New(lockFilePath)
+		if err != nil {
+			log.Error(err, "Failed to create lockFile for NMSTATE_INSTANCE_NODE_LOCK_FILE")
+			os.Exit(1)
+		}
+		for {
```

Comment:
We need also timeouts to free resources in case of lock defects.
For that at knmstate usually use https://godoc.org/k8s.io/apimachinery/pkg/util/wait

Also I have found at project that mix  mutext and context so we don't need the for loop at all https://github.com/sasha-s/go-csync.

Reply:
Not sure that I understood your point here, could you please elaborate about timeouts to free resources thing (what do you mean by lock defects?)
And also about the usage of context mutex .